src/goto-programs/goto_contractor.cpp

```diff
@@ -13,27 +18,74 @@ void goto_contractort::get_contractors(goto_functionst goto_functions)
   for(const auto &ins : function->second.body.instructions)
   {
     if(ins.is_assert())
+    {
       contractors.add_contractor(
-        create_contractor_from_expr2t(ins.guard), ins.location_number);
+        parser.parse(ins.guard), ins.location_number);
+    }
     else if(
       ins.is_function_call() &&
       is_symbol2t(to_code_function_call2t(ins.code).function) &&
       to_symbol2t(to_code_function_call2t(ins.code).function)
           .get_symbol_name() == "c:@F@__VERIFIER_assert")
       contractors.add_contractor(
-        create_contractor_from_expr2t(
+        parser.parse(
           to_code_function_call2t(ins.code).operands[0]),
         ins.location_number);
   }
 }
 
-void goto_contractort::get_intervals(goto_functionst goto_functions)
+void goto_contractort::get_intervals(
+  goto_functionst goto_functions,
+  const namespacet &ns)
 {
-  auto function = goto_functions.function_map.find("c:@F@main");
+  ait<interval_domaint> interval_analysis;
 
-  for(const auto &ins : function->second.body.instructions)
-    if(ins.is_assume())
-      parse_intervals(ins.guard);
+  interval_analysis(goto_functions, ns);
+
+  std::ostringstream oss;
+  interval_analysis.output(goto_functions, oss);
+
+  std::ostringstream oss2;
+
+  //Forall_goto_functions(f_it, goto_functions)
+  auto f_it = goto_functions.function_map.find("c:@F@main");
+  {
+    Forall_goto_program_instructions(i_it, f_it->second.body)
+    {
+      if(
+        i_it->is_assert() ||
+        (i_it->is_function_call() &&
+         is_symbol2t(to_code_function_call2t(i_it->code).function) &&
+         to_symbol2t(to_code_function_call2t(i_it->code).function)
+             .get_symbol_name() == "c:@F@__VERIFIER_assert"))
+      {
+        auto it = map.var_map.begin();
+        while(it != map.var_map.end())
+        {
+          auto var_name = to_symbol2t(it->second.getSymbol()).get_symbol_name();
+          auto new_interval = interval_analysis[i_it].get_int_map()[var_name];
+          if(!new_interval.is_top())
+          {
+            auto lb = new_interval.get_lower().to_int64();
+            auto ub = new_interval.get_upper().to_int64();
+            auto i = it->second.getInterval();
+
+            std::cout << "at location: " << i_it->location_number
```

Comment:
We need to use our methods to report the log.

Reply:
You can use `log_debug("contractor", "at location: {} Var {} this out {} should be: {}", ...)`.

Reply:
Done.